### ISSUE TITLE
Bug - Core 4.4.5 - Crash due to long check output, probably same as #432

### ISSUE THREAD

**Issue by attilatibor-pakozdi** [NagiosEnterprises/nagioscore#719]
Probably same as issue #432 

About to migrate from 3.5 to 4.4.5 when encountered frequent crashes.

Narrowed down to certain checks which return long output sometimes. After some digging and testing, here is what we have:

**Debug trace**

> Nagios 4.4.5 starting... (PID=18034)
> Local time is Wed Jan 15 14:40:55 UTC 2020
> wproc: Successfully registered manager as @wproc with query handler
> Detaching after fork from child process 18035.
> Detaching after fork from child process 18036.
> Detaching after fork from child process 18037.
> Detaching after fork from child process 18038.
> Detaching after fork from child process 18039.
> Detaching after fork from child process 18040.
> Detaching after fork from child process 18041.
> Detaching after fork from child process 18042.
> Detaching after fork from child process 18043.
> Detaching after fork from child process 18044.
> Detaching after fork from child process 18045.
> Detaching after fork from child process 18046.
> Detaching after fork from child process 18047.
> Detaching after fork from child process 18048.
> Detaching after fork from child process 18049.
> Detaching after fork from child process 18050.
> Detaching after fork from child process 18051.
> Detaching after fork from child process 18052.
> Detaching after fork from child process 18053.
> Detaching after fork from child process 18054.
> Detaching after fork from child process 18055.
> Detaching after fork from child process 18056.
> Detaching after fork from child process 18057.
> Detaching after fork from child process 18058.
> wproc: Registry request: name=Core Worker 18035;pid=18035
> wproc: Registry request: name=Core Worker 18036;pid=18036
> wproc: Registry request: name=Core Worker 18037;pid=18037
> wproc: Registry request: name=Core Worker 18038;pid=18038
> wproc: Registry request: name=Core Worker 18039;pid=18039
> wproc: Registry request: name=Core Worker 18040;pid=18040
> wproc: Registry request: name=Core Worker 18041;pid=18041
> wproc: Registry request: name=Core Worker 18042;pid=18042
> wproc: Registry request: name=Core Worker 18043;pid=18043
> wproc: Registry request: name=Core Worker 18044;pid=18044
> wproc: Registry request: name=Core Worker 18045;pid=18045
> wproc: Registry request: name=Core Worker 18046;pid=18046
> wproc: Registry request: name=Core Worker 18047;pid=18047
> wproc: Registry request: name=Core Worker 18048;pid=18048
> wproc: Registry request: name=Core Worker 18049;pid=18049
> wproc: Registry request: name=Core Worker 18051;pid=18051
> wproc: Registry request: name=Core Worker 18050;pid=18050
> wproc: Registry request: name=Core Worker 18052;pid=18052
> wproc: Registry request: name=Core Worker 18053;pid=18053
> wproc: Registry request: name=Core Worker 18055;pid=18055
> wproc: Registry request: name=Core Worker 18054;pid=18054
> wproc: Registry request: name=Core Worker 18056;pid=18056
> wproc: Registry request: name=Core Worker 18057;pid=18057
> wproc: Registry request: name=Core Worker 18058;pid=18058
> Failed to send kvvec struct to master: Broken pipe
> Warning: Service performance file processing command 'process-service-perfdata-for-nagiosgraph' was not found - service performance data file will not be processed!
> Detaching after fork from child process 18060.
> Successfully launched command file worker with pid 18060
> Warning: Return code of 7 for check of service 'Nagiosbug1' on host 'localhost' was out of bounds.
> wproc: iocache_capacity() is -1048576 for worker Core Worker 18037.
> wproc: iocache_capacity() is -2097152 for worker Core Worker 18038.
> wproc: iocache_capacity() is -2097152 for worker Core Worker 18039.
> Warning: Return code of 7 for check of service 'Nagiosbug13' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug14' on host 'localhost' was out of bounds.
> wproc: iocache_capacity() is -1048576 for worker Core Worker 18042.
> wproc: iocache_capacity() is -4194304 for worker Core Worker 18042.
> Warning: Return code of 7 for check of service 'Nagiosbug16' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug17' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug18' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug19' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug2' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug20' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug21' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug22' on host 'localhost' was out of bounds.
> Warning: Return code of 7 for check of service 'Nagiosbug23' on host 'localhost' was out of bounds.
> wproc: iocache_capacity() is -1048576 for worker Core Worker 18053.
> wproc: iocache_capacity() is -4194304 for worker Core Worker 18053.
> wproc: iocache_capacity() is -2097152 for worker Core Worker 18055.
> *** Error in `/aot/swissbud/nagios/bin/nagios': realloc(): invalid old size: 0x00007ffff60f3010 ***
> ======= Backtrace: =========
> /lib64/libc.so.6(+0x7f804)[0x7ffff7586804]
> /lib64/libc.so.6(+0x84bf1)[0x7ffff758bbf1]
> /lib64/libc.so.6(realloc+0x1d2)[0x7ffff758d1a2]
> /aot/swissbud/nagios/bin/nagios(iocache_resize+0x3c)[0x473c4c]
> /aot/swissbud/nagios/bin/nagios(iocache_read+0x9d)[0x47405d]
> /aot/swissbud/nagios/bin/nagios[0x41929f]
> /aot/swissbud/nagios/bin/nagios(iobroker_poll+0x83)[0x4746d3]
> /aot/swissbud/nagios/bin/nagios(event_execution_loop+0x221)[0x433551]
> /aot/swissbud/nagios/bin/nagios(main+0xb99)[0x413bc9]
> /lib64/libc.so.6(__libc_start_main+0xf5)[0x7ffff7529545]
> /aot/swissbud/nagios/bin/nagios[0x4142fb]
> ======= Memory map: ========
> 00400000-004ab000 r-xp 00000000 fd:08 100901364                          /aot/swissbud/nagios-4.4.5/bin/nagios
> 006aa000-006ab000 r--p 000aa000 fd:08 100901364                          /aot/swissbud/nagios-4.4.5/bin/nagios
> 006ab000-006ae000 rw-p 000ab000 fd:08 100901364                          /aot/swissbud/nagios-4.4.5/bin/nagios
> 006ae000-011c2000 rw-p 00000000 00:00 0                                  [heap]
> 7fffec000000-7fffec021000 rw-p 00000000 00:00 0
> 7fffec021000-7ffff0000000 ---p 00000000 00:00 0
> 7ffff27e8000-7ffff2fe9000 rw-p 00000000 00:00 0
> 7ffff2fe9000-7ffff37ea000 rw-p 00000000 00:00 0
> 7ffff37ea000-7ffff3feb000 rw-p 00000000 00:00 0
> 7ffff3feb000-7ffff47ec000 rw-p 00000000 00:00 0
> 7ffff47ec000-7ffff4fed000 rw-p 00000000 00:00 0
> 7ffff4fed000-7ffff57ee000 rw-p 00000000 00:00 0
> 7ffff5df0000-7ffff61f4000 rw-p 00000000 00:00 0
> 7ffff63f6000-7ffff6e00000 rw-p 00000000 00:00 0
> 7ffff6f01000-7ffff7103000 rw-p 00000000 00:00 0
> 7ffff71f0000-7ffff7205000 r-xp 00000000 fd:00 5907686                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
> 7ffff7205000-7ffff7404000 ---p 00015000 fd:00 5907686                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
> 7ffff7404000-7ffff7405000 r--p 00014000 fd:00 5907686                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
> 7ffff7405000-7ffff7406000 rw-p 00015000 fd:00 5907686                    /usr/lib64/libgcc_s-4.8.5-20150702.so.1
> 7ffff7406000-7ffff7507000 rw-p 00000000 00:00 0
> 7ffff7507000-7ffff76ca000 r-xp 00000000 fd:00 4202780                    /usr/lib64/libc-2.17.so
> 7ffff76ca000-7ffff78ca000 ---p 001c3000 fd:00 4202780                    /usr/lib64/libc-2.17.so
> 7ffff78ca000-7ffff78ce000 r--p 001c3000 fd:00 4202780                    /usr/lib64/libc-2.17.so
> 7ffff78ce000-7ffff78d0000 rw-p 001c7000 fd:00 4202780                    /usr/lib64/libc-2.17.so
> 7ffff78d0000-7ffff78d5000 rw-p 00000000 00:00 0
> 7ffff78d5000-7ffff78d7000 r-xp 00000000 fd:00 4202788                    /usr/lib64/libdl-2.17.so
> 7ffff78d7000-7ffff7ad7000 ---p 00002000 fd:00 4202788                    /usr/lib64/libdl-2.17.so
> 7ffff7ad7000-7ffff7ad8000 r--p 00002000 fd:00 4202788                    /usr/lib64/libdl-2.17.so
> 7ffff7ad8000-7ffff7ad9000 rw-p 00003000 fd:00 4202788                    /usr/lib64/libdl-2.17.so
> 7ffff7ad9000-7ffff7bda000 r-xp 00000000 fd:00 5244942                    /usr/lib64/libm-2.17.so
> 7ffff7bda000-7ffff7dd9000 ---p 00101000 fd:00 5244942                    /usr/lib64/libm-2.17.so
> 7ffff7dd9000-7ffff7dda000 r--p 00100000 fd:00 5244942                    /usr/lib64/libm-2.17.so
> 7ffff7dda000-7ffff7ddb000 rw-p 00101000 fd:00 5244942                    /usr/lib64/libm-2.17.so
> 7ffff7ddb000-7ffff7dfd000 r-xp 00000000 fd:00 4195308                    /usr/lib64/ld-2.17.so
> 7ffff7ec9000-7ffff7fee000 rw-p 00000000 00:00 0
> 7ffff7ff5000-7ffff7ffa000 rw-p 00000000 00:00 0
> 7ffff7ffa000-7ffff7ffc000 r-xp 00000000 00:00 0                          [vdso]
> 7ffff7ffc000-7ffff7ffd000 r--p 00021000 fd:00 4195308                    /usr/lib64/ld-2.17.so
> 7ffff7ffd000-7ffff7ffe000 rw-p 00022000 fd:00 4195308                    /usr/lib64/ld-2.17.so
> 7ffff7ffe000-7ffff7fff000 rw-p 00000000 00:00 0
> 7ffffffde000-7ffffffff000 rw-p 00000000 00:00 0                          [stack]
> ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
> 
> Program received signal SIGABRT, Aborted.
> 0x00007ffff753d377 in raise () from /lib64/libc.so.6

The following check triggers it :
```
MAIN:{
  my $string = 'A' x 1024;
  $string = $string x 1024;
  $string = $string x 8; # 8 MBytes
  print $string;
  exit 0;
}
```

**Workaround**: Our solution is to make sure we do not send this much data from our checks, but digging through 10+ years of scripts is time consuming.

**Comment by sawolf** [comment]
Thanks for reporting this.

**Comment by Kilvador** [comment]
https://github.com/NagiosEnterprises/nagioscore/issues/825

**Comment by flicker581** [comment]
https://github.com/NagiosEnterprises/nagioscore/pull/827

**Comment by sawolf** [comment]
I believe this was fixed in nagios 4.4.7 as part of #825/#828 - feel free to re-open this or create a new issue if you're still seeing this failure
